src/Adapter/Address/AbstractAddressHandler.php

```diff
@@ -0,0 +1,63 @@
+<?php
+/**
+ * 2007-2019 PrestaShop and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Adapter\Address;
+
+use Address;
+use PrestaShop\PrestaShop\Core\Domain\Address\Exception\AddressException;
+use PrestaShop\PrestaShop\Core\Domain\Address\Exception\AddressNotFoundException;
+use PrestaShop\PrestaShop\Core\Domain\Address\ValueObject\AddressId;
+use PrestaShopException;
+
+/**
+ * Provides reusable methods for address command/query handlers
+ */
+abstract class AbstractAddressHandler
```

Comment:
Do we need this class ? Is AbstractAddressCommandHandler not enough ?

Reply:
In current PR it might seem unnecessary, but in create/edit action use cases appears when for example getAddress method is used by both - commands and queries, but couple other methods are reused only in commands but not queries. ( I hope I made it clear :D ). In short - I tried to separate Abstract methods for command(AbstractAddressCommandHandler), queries and one common that is used by both (AbstractAddressHandler).  On the other hand, it might be enough to have one AbstractAddressHandler for all reusible methods, its your call.

Reply:
Let's start with a single AbstractAddressHandler then, and if relevant expand it into AbstractAddressCommandHandler and AbstractAddressQueryHandler